libp2p/nameresolving/dnsresolver.nim

```diff
@@ -79,6 +79,10 @@ proc getDnsResponse(
     dataStream.writeData(addr rawResponse[0], rawResponse.len)
     dataStream.setPosition(0)
     return parseResponse(dataStream)
+  except CatchableError as exc:
+    raise exc
+  except Exception as exc: # DNS library can throw anything
+    raise newException(CatchableError, "DNS query error: " & exc.msg)
```

Comment:
This should be `Defect`.

Reply:
fwiw, we usually type this as `raiseAssert exc.msg` - also, unless the dns library actuall_ raises exceptions that are not `CatchableError`, it's not a real problem

the most important thing here is that dns resolution should generally not happen automatically - that introduces a number of security issues and considerations that need to be taken into account in addition to language details like this - ie resolution should always be explicitly enabled to happen

Reply:
You have to pass a nameresolver to the switch for it to happen